jetty-util/src/main/java/org/eclipse/jetty/util/URIUtil.java

```diff
@@ -1122,6 +1122,56 @@ public static boolean hasScheme(String uri)
         return false;
     }
 
+    /**
+     * True if token is a <a href="https://www.rfc-editor.org/rfc/rfc3986">RFC3986</a> {@code reg-name}
+     *
+     * @param token the to test
+     * @return true if the token passes as a valid reg-name
+     */
+    public static boolean isRegName(String token)
```

Comment:
While "regName" is what the spec calls it, I would change it to `isValidHost()`.
I have no idea what "reg" means in this context, and I don't want to look it up (lazy), so if it's named `isValidHost()` I'll know what it is.

Reply:
Per https://www.rfc-editor.org/rfc/rfc3986

This is `reg-name` only, and not `host` as this method does not validate `IP-literal` or `IPv4address` (which is also defined in `host`)